mlrun/model_monitoring/db/tsdb/tdengine/schemas.py

```diff
@@ -148,10 +173,21 @@ def _get_records_query(
                     query.write(f"{timestamp_column} >= '{start}'" + " and ")
                 if end:
                     query.write(f"{timestamp_column} <= '{end}'")
+            full_query = query.getvalue()
+            if full_query.endswith(" and "):
+                full_query = full_query[:-5]
+        if any([limit, interval]):
+            with StringIO() as query:
+                query.write(full_query)
+                if interval:
+                    query.write(f" INTERVAL({interval})")
+                if sliding_window:
+                    query.write(f" SLIDING({sliding_window})")
+                if limit:
+                    query.write(f" LIMIT {limit}")
                 full_query = query.getvalue()
-                if full_query.endswith(" and "):
-                    full_query = full_query[:-5]
-            return full_query + ";"
+
+        return full_query + ";"
```

Comment:
Same comment as above. The whole point of using `StringIO` is to avoid string concatenation